src/store/getters/state-getters.ts

```diff
@@ -502,12 +508,34 @@ export const hasConversionDataChanged = (state: StateIF): boolean => {
 
 /** Whether all correction sections are valid. */
 export const isFilingValid = (state: StateIF): boolean => {
-  // NB: Define Company and Agreement Type don't have a "valid" state --
-  //     they don't allow saving an invalid state to the store.
-  return (state.stateModel.peopleAndRoles.valid &&
-    state.stateModel.detail.valid &&
-    state.stateModel.certifyState.valid &&
-    state.stateModel.staffPaymentStep.valid)
+  if (isBenIaCorrectionFiling(state)) {
+    // NB: Define Company and Agreement Type don't have a "valid" state --
+    //     they don't allow saving an invalid state to the store.
+    return (
+      state.stateModel.peopleAndRoles.valid &&
+      state.stateModel.detail.valid &&
+      state.stateModel.certifyState.valid &&
+      state.stateModel.staffPaymentStep.valid
+    )
+  }
+  if (isFirmCorrectionFiling(state)) {
+    // *** FUTURE: should check the following:
```

Comment:
To finish when future tickets look at validation.

This works for now but might not catch all scenarios.